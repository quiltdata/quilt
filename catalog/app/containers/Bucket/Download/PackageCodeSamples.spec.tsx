import * as React from 'react'
import renderer from 'react-test-renderer'

import PackageCodeSamples from './PackageCodeSamples'

jest.mock(
  './Code',
  () =>
    ({ label, help, lines }: { label: string; help: string; lines: string[] }) => (
      <dl key={label} data-help={help}>
        <dt>{label}:</dt>
        {lines.map((l, i) => (
          <dd key={`${l}_${i}`}>{l}</dd>
        ))}
      </dl>
    ),
)

describe('containers/Bucket/CodeSamples/Package', () => {
  it('renders catalog property', () => {
    const props = {
      bucket: 'bucket',
      name: 'name',
      hash: 'hash',
      hashOrTag: 'tag',
      path: 'path',
      catalog: 'catalog',
    }
    const tree = renderer.create(<PackageCodeSamples {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
