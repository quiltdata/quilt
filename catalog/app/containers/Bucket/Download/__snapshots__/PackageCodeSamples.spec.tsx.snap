// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Bucket/CodeSamples/Package renders catalog property 1`] = `
<div>
  <dl
    data-help="https://docs.quiltdata.com/quilt-python-sdk-developers/api-reference/package#package.browse"
  >
    <dt>
      Browse using Quilt3 Python API
      :
    </dt>
    <dd>
      import quilt3 as q3
    </dd>
    <dd>
      q3.Package.browse("name", top_hash="hash", registry="s3://bucket")
    </dd>
  </dl>
  <dl
    data-help="https://docs.quiltdata.com/quilt-python-sdk-developers/api-reference/package#package.install"
  >
    <dt>
      Install using Quilt3 Python API
      :
    </dt>
    <dd>
      import quilt3 as q3
    </dd>
    <dd>
      q3.Package.install("name", path="path", top_hash="hash", registry="s3://bucket", dest=".")
    </dd>
  </dl>
  <dl
    data-help="https://docs.quiltdata.com/quilt-python-sdk/api-reference/cli#install"
  >
    <dt>
      Install using Quilt3 CLI
      :
    </dt>
    <dd>
      quilt3 install "name" --path "path" --top-hash hash --registry s3://bucket --dest .
    </dd>
  </dl>
</div>
`;
